flashcard_runner.rb

```diff
@@ -0,0 +1,52 @@
+require './lib/card'
+require './lib/deck'
+require './lib/round'
+require './lib/turn'
+require './lib/card_generator.rb'
+
+def start
+    # card1 = Card.new("What is your name?", "Sir Lancelot of Camelot", :Identity)
+    # card2 = Card.new("What is your quest?", "To seek the Holy Grail", :Identity)
+    # card3 = Card.new("What is the capital of Assyria?", "Assur", :Geography)
+    # card4 = Card.new("What is the airspeed velocity of an unladen swallow?", "24mph", :Ornithology)
+    # cards = [card1, card2, card3, card4]
+
+    cards = CardGenerator.new("cards.txt").cards
+
+    deck = Deck.new(cards)
+
+    round = Round.new(deck)
+
+    puts "Welcome! You're playing with #{deck.count} cards."
+    puts "-------------------------------------------------"
+    
+    deck.count.times do |card_number|
+        puts "This is card number #{card_number + 1} out of #{deck.count}."
+        puts "Question: #{round.current_card.question}"
+        user_input = gets.chomp
+        puts round.take_turn(user_input).feedback 
+    end
+
+    puts "****** Game over! ******"
+    puts "You had #{round.number_correct} out of #{deck.count} for a total score of #{round.percent_correct.to_i}%."
+    
+    # Start with an empty array...
+    categories = []
+    
+    # Visit each card
+    deck.cards.each do |card|
+        # if it hasn't been added already...
+        unless categories.include?(card.category)
+            # add its category
+            categories << card.category 
+        end
+    end
+
+    categories.each do |category|
+        puts "#{category} - #{round.percent_correct_by_category(category).to_i}% correct"
+    end
+
+end
+
+start 
```

Comment:
At this point in your developer career, I'm happy with what you were able to produce, and how you were able to get your game to run via this runner file. In the future, we'll challenge you to store as little code as possible outside of a class. Right now, you have code in this file that is not in a class. So, if you were to follow that convention, you could have a line of code in your runner file that was something like Game.start. Where Game is a new class that you created for the purpose of prepping the objects, and running the game. And start was a method that kicked off the flow of the game. The idea is that the flow of the game is stored in methods in that game class.